protocols/api/src/main/java/org/apache/james/protocols/api/OidcSASLConfiguration.java

```diff
@@ -0,0 +1,46 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.protocols.api;
+
+import java.net.URL;
+
+public class OidcSASLConfiguration {
+    private final URL jwksURL;
+    private final String claim;
+    private final URL oidcConfigurationURL;
```

Comment:
From my work on SASL for SMTP I do remember a `private final String scope;` field was handy...

Reply:
I haven't figured out what the scope's purpose is yet. Delegate?

Reply:
Honestly, I also don't understand why we need `oidcConfigurationURL` too? Just for more information? I guess

Reply:
We need `oidcConfigurationURL` as it need to be returned upon errors to the client.